### DISCUSSION TITLE
Waiting for server response > 2s

### DISCUSSION THREAD

**Discussion by fxlelouarn** [swellstores/community#253] [Help] [answered]
Hi there. We're experiencing significant delays when calling API endpoints (2-3 seconds). I've read in another thread that it might be because of the API infrastructure and that work is being undertaken to move everything to a single cloud provider. Do you have any idea as when this move will be completed? How high is this in Swell's priorities? Are we talking about weeks, months, quarters, etc.?

I've attached screenshots of the delays we're experiencing.

<img width="1334" alt="Capture d’écran 2023-05-11 à 16 33 33" src="https://github.com/swellstores/community/assets/115778739/f387f7be-6a9e-4eb5-9ff4-fd90d002519c">

Thanks for your answer.
FX.

**Comment by greph** [comment, marked as answer]
Hi @fxlelouarn , I don't have a timeline for you on the cloud infra updates but am wondering if some caching strategies could be used here in order to reduce wait times. For example, the cart session could be managed locally before submitting as an order. Likewise for products an [ISR strategy](https://nextjs.org/docs/pages/building-your-application/data-fetching/incremental-static-regeneration) could be used for-demand revlaidation, in order to update items as changes are made.

Out of curiosity, are you using Next.js for your storefront?

**Reply by logeshswell** [reply]
Secondly, another reason for the slower response time could be API throttling on excessive usage. The Standard plan has higher allowances, and Enterprise plans have no API limits.
For more details, please refer to https://www.swell.is/pricing

**Reply by igoraugustynski** [reply]
Yeah no something is seriously wrong with your infrastructure..
<img width="1726" alt="Screenshot 2023-06-15 at 22 28 14" src="https://github.com/swellstores/community/assets/94485687/06bd404e-4c3a-4ea8-aee9-9681ba1a2c4d">

**Reply by ericingram** [reply]
We're working on it

**Comment by mewa** [comment]
Very slow page loads is something we've been battling ourselves so we decided to cache API calls where possible. So far our results have been pretty good, bringing down most calls to ~100ms range.

Most importantly, the store loads visibly faster and feels much snappier. We're hoping to see an uptick in conversion rate soon.

For the record, here's our page load before and after:
<img width="628" alt="Page load without caching" src="https://github.com/swellstores/community/assets/7347194/2cbb422f-2e35-47b4-ada2-d3998df9bd35">

<img width="608" alt="Page load with caching" src="https://github.com/swellstores/community/assets/7347194/f07417a8-492f-4552-aba4-826f1a8f22d5">

If you're interested in speeding up your store let me know through our form at [zippityzapp.com](https://zippityzapp.com).

**Comment by ericingram** [comment]
We've made major improvements recently and are seeing request times improved across the board.

@igoraugustynski @fxlelouarn Let us know if you can confirm, and thank you for your patience

**Reply by mewa** [reply]
FWIW, it does seem to have improved since I've first observed it and started working on the cache. We used to see lots of double-digit response times, whereas now they seem to hover in the low single-digit range (although I do wonder about the tail latencies).

It's still fairly slow for an ecommerce system, especially considering that Mongo is on the backend and a lot of the data is denormalized already.

I think there is some low-hanging fruit to be had here, such as the call to `/api/settings/all`, which blocks all other requests. Layered with a 1-2s call to get the products means the website only becomes interactive after a minimum of 2-3s.

**Reply by ericingram** [reply]
@mewa thanks for the details on what you see. We have another couple of improvements coming, including API-level caching and then edge-caching. These should improve response times further by an order of magnitude.

**Reply by igoraugustynski** [reply]
I can confirm that it seems much faster now, no double digit response times - pretty much every API call is < 400ms. Great job guys!

**Reply by fxlelouarn** [reply]
@ericingram Thanks for the update and congrats for the speed improvement. We weren't able to wait for it as our new checkout tunnel was directly impacted by the latencies. We decided to remove Swell dependencies and replace them with a Medusa instance we host here in Europe. The change took us approximately 3 person-weeks and our API calls now have acceptable delays. We'll keep following Swell's evolution.

**Reply by ericingram** [reply]
Totally understandable @fxlelouarn! We'll have some exciting news to share in the next few months

**Reply by igoraugustynski** [reply]
<img width="1172" alt="Screenshot 2023-08-01 at 11 50 31" src="https://github.com/swellstores/community/assets/94485687/146974cf-161b-43f4-832c-1d618c56bd0e">
Never mind, it's still as bad as it was before.
